### ISSUE TITLE
Training Degrades After ~3000 Steps on NeRSemble Multi-View Dataset

### ISSUE THREAD

**Issue by KEH0T0** [gapszju/RGBAvatar#8]
I switched the code to use NeRSemble’s multi-view dataset and ran training. Since it contains around 5,000 images, perhaps due to the large data volume, training with train_offline seems to proceed normally up to about 2,000 steps, but then the results begin to deteriorate from around 3,000 steps onward. Could you share your thoughts on whether this is a bug in the code or a hyperparameter mismatch? I’ve attached the output screenshots/results below.

![Image](https://github.com/user-attachments/assets/7c19f96e-12c7-4a0c-be26-ab8c17b63c56)
![Image](https://github.com/user-attachments/assets/9977bd44-72cc-435d-a0bc-d3b97c1d3b1c)

**Comment by LinzhouLi** [comment]
Hi, thanks for your feedback on multi-view avatar reconstruction!

Just to clarify, the training of blendshape bases starts after 3000 iterations (controlled by the blend_start_iter parameter in the config file). However, I don't believe this is the root cause of the issue you're seeing.

Do you mean that the training fails to converge even after all 50k iterations? If so, you might try reducing the learning rate for the Gaussian parameters, as that can help stabilize training.

Could you share more details about your training setup, such as the batch size and any modifications you made to the default configuration? This will help me better understand and diagnose the problem.

**Comment by KEH0T0** [comment]
Thanks for quick response. 
I’ve already tried adjusting the control `blend_start_iter` parameter, but it didn’t work as you described.

Here is my config.
 ```
      dataset:
        pin_memory: True
        use_shape_weight: True
        use_pose_weight: True
      
      model:
        tex_size: 256
        num_basis_in: 127
        num_basis_blend: 50
        use_blend: True
        use_weight_proj: True
        use_mlp_proj: True
      
        init_scaling: 0.0008
        init_opacity: 0.5
      
      train:
        batch_size: 1
        iteration: 300_000
      
      recon:
        blend_start_iter: 18000
        use_fast_forward: True
        random_bg_color: True
        bg_color: [0.0, 0.0, 0.0]
    
        scene_extent: 1.0
        position_lr: 0.0008
        position_lr_final: 0.000008

        position_lr_delay_mult: 0.01
        position_lr_max_steps: 300_000
        opacity_lr: 0.25
        scaling_lr: 0.025
        rotation_lr: 0.005
        feature_lr: 0.0125
    
        position_b_lr_scale: 0.05
        rotaion_b_lr_scale: 0.5
        feature_b_lr_scale: 0.5
    
        weight_module_lr: 0.001
    
        lambda_l1: 1.0
        lambda_ssim: 0.0
        lambda_lpips: 0.0
        lambda_alpha: 0.0
        lambda_sparsity: 0.0
        lambda_orth: 0.0
 ```

I also scaled the learning rate for all Gaussian parameters by 0.1.
I think it’s working, but I’m still seeing blurry and floating artifacts. If I tune the hyperparameters further, do you think I could achieve finer details? or Is there another problem i have to solve?


![Image](https://github.com/user-attachments/assets/349b5544-3d54-497f-8b08-e0034d9fbbd9)
![Image](https://github.com/user-attachments/assets/4fb07534-cd5a-45a5-97e0-bdf6300037ad)

**Comment by LinzhouLi** [comment]
I notice that you use batch_size=1 during training. The learning rate should be adjusted according to the training batch size. I suggest trying the learning rate settings from the original Gaussian Splatting repo for batch size 1:
```
scene_extent: 1.0
position_lr: 0.00016
position_lr_final: 0.0000016
position_lr_delay_mult: 0.01
position_lr_max_steps: 30_000
opacity_lr: 0.025
scaling_lr: 0.005
rotation_lr: 0.001
feature_lr: 0.0025
```

Regarding the floating artifacts in your first figure, they might be caused by pixel values exceeding the [0, 1] range. Please make sure to clamp your output image tensor within this range before saving or visualizing. Moreover, in my experience, these kinds of artifacts often occur when the model hasn’t fully converged. You might try training for more iterations — for reference, GaussianAvatars trained each avatar for up to 600,000 iterations.
